mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -494,37 +494,49 @@ private static MlflowHostCredsProvider getHostCredsProviderFromTrackingUri(Strin
   }
 
   /**
-   * Upload the given local file to the run's root artifact directory. For example,
+   * Upload the given local file or directory to the run's root artifact directory. For example,
    *
    *   <pre>
    *   logArtifact(runId, "/my/localModel")
    *   listArtifacts(runId) // returns "localModel"
    *   </pre>
    *
    * @param runId Run ID of an existing MLflow run.
-   * @param localFile File to upload. Must exist, and must be a simple file (not a directory).
+   * @param localFile File or directory to upload. Must exist.
    */
   public void logArtifact(String runId, File localFile) {
-    getArtifactRepository(runId).logArtifact(localFile);
+    if (localFile.isDirectory()) {
```

Comment:
Should we also update Java tests to verify that we can pass a directory to `logArtifact`?